e2e/validation.go

```diff
@@ -0,0 +1,123 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/gomega"
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	timeout  = time.Second * 10
+	interval = time.Millisecond * 100
+)
+
+// function to validate LVMVolume group.
+func ValidateLVMvg() error {
+	lvmVolumeGroupName := "vg1"
+	lvmVolumeGroupNamespace := InstallNamespace
+	lvmVG := lvmv1alpha1.LVMVolumeGroup{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: lvmVolumeGroupName, Namespace: lvmVolumeGroupNamespace}, &lvmVG)
+		return err == nil
+	}, timeout, interval).Should(BeTrue())
+
+	return nil
+}
+
+// function to validate storage class.
+func ValidateStorageClass() error {
+	storageClassName := "odf-lvm-vg1"
+	storageClassNamespace := InstallNamespace
+	sc := storagev1.StorageClass{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: storageClassName, Namespace: storageClassNamespace}, &sc)
+		return err == nil
+	}, timeout, interval).Should(BeTrue())
+
+	return nil
+}
+
+// function to validate CSI Driver.
+func ValidateCSIDriver() error {
+	csiDriverName := "topolvm.cybozu.com"
+	csiDriverNamespace := InstallNamespace
+	cd := storagev1.CSIDriver{}
+
+	Eventually(func() bool {
+		err := DeployManagerObj.GetCrClient().Get(context.TODO(), types.NamespacedName{Name: csiDriverName, Namespace: csiDriverNamespace}, &cd)
+		return err == nil
+	}, timeout, interval).Should(BeTrue())
+
+	return nil
+}
+
+// function to validate Daemonset(VG Manager resource).
+func ValidateDaemonset() error {
```

Comment:
```suggestion
func ValidateTopolvmNode() error {
```

Reply:
The vgmanager daemonset also needs to be checked.

Reply:
Updated